src/App.js

```diff
@@ -67,7 +79,11 @@ function App() {
         <Profile results={quizResult} products={shelfState} />
       </Route>
       <Route exact path="/shelf">
-        <ShelfPage products={shelfState} onProductClick={replaceOnShelf} />
+        <ShelfPage
+          userProducts={shelfState}
+          allProducts={allProducts}
+          onProductClick={replaceOnShelf}
+        />
       </Route>
```

Comment:
These Routes look good! I think if you wanted to cover error handling for an incorrect url, all you'd have to do is add one final Route at the bottom for path='/*' that would catch any other url that's not /, /quiz, /profile, or /shelf(since you don't have any dynamic routes). Then you would just have that route render an error component!